src/main/java/io/cryostat/discovery/PluginInfo.java

```diff
@@ -0,0 +1,141 @@
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.discovery;
+
+import java.net.URI;
+import java.util.UUID;
+
+import javax.persistence.Column;
+import javax.persistence.Convert;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+import com.vladmihalcea.hibernate.type.json.JsonBinaryType;
+import org.hibernate.annotations.GenericGenerator;
+import org.hibernate.annotations.Type;
+import org.hibernate.annotations.TypeDef;
+
+@Entity
+@TypeDef(name = "jsonb", typeClass = JsonBinaryType.class)
+public class PluginInfo {
+
+    @Id
+    @GeneratedValue(generator = "UUID")
+    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
+    private UUID id;
+
+    @Column(unique = true, nullable = false)
+    private String realm;
+
+    @Column
+    @Convert(converter = UriConverter.class)
+    private URI callback;
+
+    @Type(type = "jsonb")
+    @Column(nullable = false, columnDefinition = "jsonb")
+    private String subtree;
+
+    PluginInfo() {}
+
+    PluginInfo(String realm, URI callback, String subtree) {
+        this.realm = realm;
+        this.callback = callback;
+        this.subtree = subtree;
+    }
+
+    public UUID getId() {
+        return id;
+    }
+
+    public String getRealm() {
+        return realm;
+    }
+
+    public URI getCallback() {
+        return callback;
+    }
+
+    public String getSubtree() {
+        return subtree;
+    }
+
+    public void setId(UUID id) {
+        this.id = id;
+    }
+
+    public void setRealm(String realm) {
+        this.realm = realm;
+    }
+
+    public void setCallback(URI callback) {
+        this.callback = callback;
+    }
+
+    public void setSubtree(String subtree) {
+        this.subtree = subtree;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
I'm curious at the significance of this hashCode function. Does the prime make sure of the uniqueness of a PluginInfo?

Reply:
It's auto-generated.

https://stackoverflow.com/questions/3613102/why-use-a-prime-number-in-hashcode

Reply:
Ahh, I see. Interesting.